tools/CreateDebugNuget.ps1

```diff
@@ -0,0 +1,284 @@
+# CreateDebugNuget.ps1
+#
+# This script will use the local reactive-domain repo
+# To create local debug reactive-domain nugets
+# reactive-domain debug build should be done before running this script
+# It will copy the debug nuspec files and the bld directory from the reactive-doain repo
+# to a temp directory. From There it will pack the nugets 
+# The resulting .nupkg files will be in the tools dir of the package-and-deployment repo
+
+# args[0]: - the path to the reactive-domain repo
+#           Note: If no valid path is passed in the script will ask for a valid path
+
+
+$configuration = "Release"
+$nuspecExtension = ".nuspec"
+$masterString = "master"
+$ReactiveDomainRepo = ""
+
+if ($args[0] -eq $null )
+{
+	$ReactiveDomainRepo = Read-Host -Prompt 'Input valid path to reactiveDomain repo'
+}
+else
+{
+	if (Test-Path -Path $args[0])
+	{
+		$ReactiveDomainRepo = $args[0]
+	}
+	else
+	{
+		$ReactiveDomainRepo = Read-Host -Prompt 'Input valid path to reactiveDomain repo'
+	}
+}
+
+Write-Host ("*********************   Begin Create Nuget script   **************************************")  
+
+Write-Host ("Copy ReactiveDomain build folder and nuspec files to a temp directory")
+
+$TempNum = Get-Random -Minimum 100 -Maximum 1000
+$TempDir = Join-Path $env:temp $TempNum.ToString()
+$buildDir = Join-Path $ReactiveDomainRepo "bld"
+$sourceDir = Join-Path $ReactiveDomainRepo "src"
+$tempBuildDir = Join-Path $TempDir "bld"
+$tempSourceDir = Join-Path $TempDir "src"
+New-Item -ItemType "directory" -Path $tempSourceDir
+Copy-Item -Path $buildDir -Destination $tempBuildDir -Recurse
+
+$sourceRDNuspec = Join-Path $sourceDir "ReactiveDomain.Debug.nuspec"
+$sourceRDTestNuspec = Join-Path $sourceDir "ReactiveDomain.Testing.Debug.nuspec"
+$sourceRDUINuspec = Join-Path $sourceDir "ReactiveDomain.UI.Debug.nuspec"
+$sourceRDUITestNuspec = Join-Path $sourceDir "ReactiveDomain.UI.Testing.Debug.nuspec"
+
+$ReactiveDomainNuspec = Join-Path $tempSourceDir "ReactiveDomain.Debug.nuspec"
+$ReactiveDomainTestingNuspec = Join-Path $tempSourceDir "ReactiveDomain.Testing.Debug.nuspec"
+$ReactiveDomainUINuspec = Join-Path $tempSourceDir "ReactiveDomain.UI.Debug.nuspec"
+$ReactiveDomainUITestingNuspec = Join-Path $tempSourceDir "ReactiveDomain.UI.Testing.Debug.nuspec"
+
+Copy-Item $sourceRDNuspec -Destination $ReactiveDomainNuspec
+Copy-Item $sourceRDTestNuspec -Destination $ReactiveDomainTestingNuspec
+Copy-Item $sourceRDUINuspec -Destination $ReactiveDomainUINuspec
+Copy-Item $sourceRDUITestNuspec -Destination $ReactiveDomainUITestingNuspec
+
+Write-Host ("Powershell script location is " + $PSScriptRoot)
+
+$RDMajor = Get-Random -Maximum 100
+$RDMinor = Get-Random -Maximum 100
+$RDRevision = Get-Random -Maximum 100
+$RDVersion = $RDMajor.ToString() + "."+ $RDMinor.ToString() + "." + $RDRevision.ToString()

```

Comment:
This isn't right. The version should come from the buildprops file.

Reply:
It should come from the build props file on a real nuget build. For a local debug nuget, the version number doesnt matter does it? 

Remember: The developer is supposed to update the build.props to create the new official nuget, otherwise the travis build will will fail. In the debug case, using the number in build.props would create a duplicate of the current version that is in the PKI Private feed. So we would have to increment that number anyway. Instead of making calles to nuget.org to check against the version in build props, we might as well just generate a random number for the debug nuget

Reply:
I understand the desire to keep this simple, but we should at least make sure that it gives us a workable version. Perhaps the following would work:
1. Get major, minor, and revision from buildprops.
2. Append "-localNNN" to the version string where NNN is a random number.

The suffix will mark it as unstable and give you a different revision number each time (mod the vagaries of random number generators and chance).